W-21447647-global-access-policy (#1016)
mulesoft/docs-gateway: gateway/1.12/modules/ROOT/pages/policies-mcp-access-control-together.adoc

> = MCP Access Control: How the Policies Work Together

Reviewer: ```suggestion
= Configuring MCP Access Control Policies
```

= MCP Access Control: How the Policies Work Together
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images
:keywords: api gateway, flex gateway, gateway, policy, mcp, access control, tool mapping, abac

> Flex Gateway provides three MCP policies that work together to control which tools and related MCP entities are exposed and how clients see them:

Reviewer: ```suggestion
Flex Gateway provides three MCP policies that work together to control which tools and related MCP entities are exposed:
```

> * xref:policies-included-mcp-global-access.adoc[MCP Global Access]: Restricts which tools are exposed, using allowlist or blocklist rules.

Reviewer: ```suggestion
* xref:policies-included-mcp-global-access.adoc[MCP Global Access]: Restricts which tools are exposed by using allow or block rules.
```

> * xref:policies-included-mcp-tool-mapping.adoc[MCP Tool Mapping]: Renames or describes tools for your use cases.

Reviewer: ```suggestion
* xref:policies-included-mcp-tool-mapping.adoc[MCP Tool Mapping]: Renames and modifies tool descriptions.
```

* xref:policies-included-mcp-attribute-access-control.adoc[MCP Attribute-Based Access Control]: Grants each client a subset of tools based on who they are, such as Tiers, IP, headers, or claims.

> Use any of these policies alone or together. Each has a clear role, nothing is implicit, and you choose only what you need.

Reviewer: ```suggestion
```

> == Three Layers: Bottom to Top

Reviewer: ```suggestion
== How the Policies Work Together
```

Think of the three policies as layers applied in order from the gateway toward the client:

Layer 1: Restrict Which Tools You Expose:: 
Apply the MCP Global Access policy to filter the overall tool list. Define Allow and Block rules so that only the tools you want to expose are available. This filters the full tool set, like a global allowlist or blocklist, and stabilizes the list of tools that can enter your network.

Layer 2: Map Tools to Names and Descriptions::
Apply the MCP Tool Mapping policy to rename tools or change their descriptions to match your use cases. Mapping doesn't implicitly allow or block tools. If you use both MCP Global Access and MCP Tool Mapping, MCP Global Access must already allow a tool before mapping applies. If you don't need to limit which tools are exposed, use only the mapping policy.

Layer 3: Grant Per-Client Access::
Apply the MCP Attribute-Based Access Control policy to give each connecting client a subset of the filtered tool list, or the mapped list if you use Tool Mapping, based on attributes such as Tiers, IP, headers, or claims. This provides attribute-based views of the tools.

> Together, these policies achieve the same outcome as a single monolithic policy but with clearer separation: each has one responsibility and you can reuse them independently.

Reviewer: ```suggestion
```

> == When to Use Which Policy

Reviewer: I don't know if I see much difference between this and the above section. I think they could be combined better or the table could be removed.

|===
| Goal | Policy to use
| Filter or stabilize the list of tools entering your network using allowlist or blocklist rules | MCP Global Access
| Rename tools or change descriptions only, no filtering | MCP Tool Mapping
| Filter and rename, no per-client rules | MCP Global Access + MCP Tool Mapping
| Different tool sets for different clients by tier, IP, headers, or claims | MCP Attribute-Based Access Control: Apply after Global Access and Tool Mapping if you use them
| Full stack: filter globally, map names and descriptions, then restrict by client | MCP Global Access, then MCP Tool Mapping, then MCP Attribute-Based Access Control
|===

> == Policy Order

Reviewer: I think this section could also be combined with the earlier section. I think this would also help reduce some of the information.
